src/Microsoft.DotNet.Wpf/src/PresentationCore/System/Windows/Automation/Peers/AutomationPeer.cs

```diff
@@ -2312,6 +2377,16 @@ private static void Initialize()
             {
                 s_propertyInfo[AutomationElementIdentifiers.PositionInSetProperty.Id] = new GetProperty(GetPositionInSet);
             }
+
+            // HeadingLevel is an RS4+ UIAutomation property.  This will be NULL on other operating system versions.
+            if (!AccessibilitySwitches.UseNetFx472CompatibleAccessibilityFeatures && AutomationElementIdentifiers.HeadingLevelProperty != null)
```

Comment:
I thought .NET Core wasn't using switches, but obviously the code above here is doing so and you're just following suit.  But you should use a new switch named UseNet50CompatibleAccessibilityFeatures, as you're protecting features that are new in 6.0.   Adding new switches to AccessibilitySwitches is a little tricky, as it's generated from a template;  let me know if you can't figure it out.

Reply:
I don't think .NET (Core) is using switches - or at least it wasn't supposed to. 

WPF has the switch infrastructure because I introduced it for the sake of back-compat in .NET Core 3.0 - in the sense that we'd wanted to give customers who were migrating from .NET 4.x  to .NET Core 3.0 the option of either using the 'on' way of doing things, or the 'off' way of doing things for the various WPF switches (i.e., let them do whatever they used to do).

My 2c: At this point, you'll probably only stop using switches if you break the inertia and stop using switches. So consider what your 'compat switch' philosophy going forward is going to be. These types of changes are great opportunities to break away from the default-choice (adding a switch) and doing something else that's more aligned with the development practices across the board (and may very well improve overall maintainability, testing etc., and also simplify documentation of breaking-changes).

In fact, you might even consider removing most (but not all - some will likely persist forever) of these switches (there is a sample that lists all of them) in .NET 6.

Reply:
@vatsan-madhavan - I'm puzzled how you can say .NET wasn't supposed to use switches, and in the next sentence admit that you added them.

Perhaps what Vatsan means is that .NET (Core) shouldn't add any _new_ switches, and that he retained the switches that were already in 4.8 for compat.   That sounds right to me.  (Although we can't get full 4.8-compat because CLR now only supports setting switches programmatically, not by app.config or registry.)

That said, let's remove the switches in the new code.  There's not really any point for switches around the new properties anyway - if people don't want the new properties to work (a doubtful situation already), they can use .NET 5.

Reply:
> I'm puzzled how you can say .NET wasn't supposed to use switches, and in the next sentence admit that you added them.

I meant it only in the *sense* that .NET Core wasn’t supposed to use the *switches* paradigm in an *ongoing* manner for compatibility-protection because runtime-updates are no longer in-place. You’re correctly surmising that I’d intended to add this infrastructure for .NET Framework 4.8 compat (only). 

That said, individual frameworks like WindowsDesktop and teams like WPF obviously have a lot of discretion here. 

Though app.config doesn’t work in core, runtimeconfig.json ought to work (our sample shows how).

Reply:
@guimafelipe - remember to remove the Accessibility switches from these two clauses.  (Keep the null check, of course)